src/Nethermind/Nethermind.Serialization.Rlp/HeaderDecoder.cs

```diff
@@ -69,7 +68,7 @@ public class HeaderDecoder : IRlpValueDecoder<BlockHeader>, IRlpStreamDecoder<Bl
             }
 
             // if we didn't reach the end of the stream, assume we have basefee to decode
-            if (decoderContext.Position != headerCheck)
+            if (decoderContext.Position != headerCheck && decoderContext.PeekPrefixAndContentLength().ContentLength != Hash256.Size)
```

Comment:
why Hash size check? What is basefee is so big it is 32 bytes?

Reply:
Indeed, it can be 32 bytes, in theory. And it's also not needed for the tests anymore, so let's remove it. But it case we will have a chain with zero base fee in the future, we can apply such update ig.